server/leafnode_test.go

```diff
@@ -4072,3 +4099,942 @@ func TestLeafNodeInterestPropagationDaisychain(t *testing.T) {
+cluster: {
+	name: clustL
+	listen: 127.0.0.1:50557
+	routes=[nats-route://127.0.0.1:50556]
+	no_advertise: true
+}
+leafnodes:{
+	no_advertise: true
+    remotes:[{url:nats://a1:a1@127.0.0.1:%d, account: A},
+		     {url:nats://s1:s1@127.0.0.1:%d, account: SYS}]
+}
+`
+
+	for _, testCase := range []struct {
+		// which topology to pick
+		same bool
+		// If leaf server should be operational and form a Js cluster prior to joining.
+		// In this setup this would be an error as you give the wrong hint.
+		// But this should work itself out regardless
+		leafFunctionPreJoin bool
+	}{
+		{true, true},
+		{true, false},
+		{false, true},
+		{false, false}} {
+		t.Run(fmt.Sprintf("%t-%t", testCase.same, testCase.leafFunctionPreJoin), func(t *testing.T) {
+			sd1 := createDir(t, JetStreamStoreDir)
+			defer os.RemoveAll(sd1)
+			confA := createConfFile(t, []byte(fmt.Sprintf(tmplA, sd1)))
+			defer removeFile(t, confA)
+			sA, _ := RunServerWithConfig(confA)
+			defer sA.Shutdown()
+
+			sd2 := createDir(t, JetStreamStoreDir)
+			defer os.RemoveAll(sd2)
+			confB := createConfFile(t, []byte(fmt.Sprintf(tmplB, sd2)))
+			defer removeFile(t, confB)
+			sB, _ := RunServerWithConfig(confB)
+			defer sB.Shutdown()
+
+			checkClusterFormed(t, sA, sB)
+
+			c := cluster{t: t, servers: []*Server{sA, sB}}
+			c.waitOnLeader()
+
+			// starting this will allow the second remote in tmplL to successfully connect.
+			port := sB.opts.LeafNode.Port
+			if testCase.same {
+				port = sA.opts.LeafNode.Port
+			}
+			p := &proxyAcceptDetectFailureLate{acceptPort: port}
+			defer p.close()
+			lPort := p.runEx(t, true)
+
+			hint := ""
+			if testCase.leafFunctionPreJoin {
+				hint = fmt.Sprintf("extension_hint: %s", strings.ToUpper(jsNoExtend))
+			}
+
+			sd3 := createDir(t, JetStreamStoreDir)
+			defer os.RemoveAll(sd3)
+			// deliberately pick server sA and proxy
+			confLA := createConfFile(t, []byte(fmt.Sprintf(tmplLA, sd3, hint, sA.opts.LeafNode.Port, lPort)))
+			defer removeFile(t, confLA)
+			sLA, _ := RunServerWithConfig(confLA)
+			defer sLA.Shutdown()
+
+			sd4 := createDir(t, JetStreamStoreDir)
+			defer os.RemoveAll(sd4)
+			// deliberately pick server sA and proxy
+			confLB := createConfFile(t, []byte(fmt.Sprintf(tmplLB, sd4, hint, sA.opts.LeafNode.Port, lPort)))
+			defer removeFile(t, confLB)
+			sLB, _ := RunServerWithConfig(confLB)
+			defer sLB.Shutdown()
+
+			checkClusterFormed(t, sLA, sLB)
+
+			strmCfg := func(name, placementCluster string) *nats.StreamConfig {
+				if placementCluster == "" {
+					return &nats.StreamConfig{Name: name, Replicas: 1, Subjects: []string{name}}
+				}
+				return &nats.StreamConfig{Name: name, Replicas: 1, Subjects: []string{name},
+					Placement: &nats.Placement{Cluster: placementCluster}}
+			}
+			// Only after the system account is fully connected can streams be placed anywhere.
+			testJSFunctions := func(pass bool) {
+				ncA := natsConnect(t, fmt.Sprintf("nats://a1:a1@127.0.0.1:%d", sA.opts.Port))
```

Comment:
I think `defer ncA.Close()` is missing here (after this call).

Reply:
fixed (added for ncL below as well)